hat/assets/js/apps/Iaso/domains/instances/compare/components/CompareInstanceLogs.tsx

```diff
@@ -24,92 +34,107 @@ type RouterCustom = {
+    };
+
+    const {
+        data: instanceLogsDropdown,
+        isFetching: isFetchingInstanceLogs,
+        isError,
+    } = useGetInstanceLogs(instanceId);
+    const {
+        userLogA,
+        userLogB,
+        isLoading: isUserLoading,
+    } = useGetUserInstanceLog(params.logA, params.logB);
+    const { data: instanceLogsDetail, isLoading: isInstanceLogDetailLoading } =
+        useGetInstanceLogDetail(params.logA, params.logB);
+
     const classes: Record<string, string> = useStyles();
+
     const { formatMessage }: { formatMessage: IntlFormatMessage } =
         useSafeIntl();
+
     const dispatch = useDispatch();
+
     const prevPathname: string | undefined = useSelector(
         (state: State) => state.routerCustom.prevPathname,
     );
+
     // FIXME ugly fix to the back arrow bug. Caused by redirecting in useEffect. using useSkipEffectOnMount breaks the feature, so this is a workaround
     // eslint-disable-next-line no-unused-vars
     const [previous, _setPrevious] = useState<string | undefined>(prevPathname);
-
+    const [instanceLogInfos, setInstanceLogInfos] = useState({
+        logA: {
+            ...logInfos,
+        },
+        logB: {
+            ...logInfos,
+        },
+    });
     const [logAInitialValue, setLogAInitialValue] = useState<
         number | undefined
     >(undefined);
     const [logBInitialValue, setLogBInitialValue] = useState<
         number | undefined
     >(undefined);
 
-    const { instanceIds: instanceId } = params;
-
-    const { data: instanceLogsDropdown, isFetching: isFetchingInstanceLogs } =
-        useGetInstanceLogs(instanceId);
-
     const handleChange = (key, value) => {
         const newParams = {
             ...params,
             [key]: value,
         };
-        dispatch(redirectTo(baseUrls.compareInstanceLogs, newParams));
+        dispatch(redirectToReplace(baseUrls.compareInstanceLogs, newParams));
     };
 
     useEffect(() => {
         if (instanceLogsDropdown) {
-            if (params.logA === undefined && params.logB === undefined) {
-                const defaultParams = {
-                    ...params,
-                    logA: instanceLogsDropdown[0]?.value,
-                    logB: instanceLogsDropdown[1]?.value,
-                };
-                dispatch(
-                    redirectTo(baseUrls.compareInstanceLogs, defaultParams),
-                );
-            } else if (params.logA === undefined) {
-                const defaultParams = {
-                    ...params,
-                    logA: instanceLogsDropdown[0]?.value,
-                };
-                dispatch(
-                    redirectTo(baseUrls.compareInstanceLogs, defaultParams),
-                );
-            } else if (params.logB === undefined) {
-                const defaultParams = {
-                    ...params,
-                    logB: instanceLogsDropdown[1]?.value,
-                };
-
+            const defaultParams: DefaultParams = {
```

Comment:
nitpick: the name doesn't seem that fitting.

Reply:
@quang-le what do you mean exatly ? It seems explicit to me

Reply:
The whole point of your `useEffect`is to determine the values of the different entries of your `defaultParams`. They're not exactly "default" parameters then. That's a bit misleading